module/VuFind/src/VuFind/Config/PathResolverFactory.php

```diff
@@ -0,0 +1,92 @@
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Config;
+
+use Laminas\ServiceManager\Exception\ServiceNotCreatedException;
+use Laminas\ServiceManager\Exception\ServiceNotFoundException;
+use Laminas\ServiceManager\Factory\FactoryInterface;
+use Psr\Container\ContainerExceptionInterface as ContainerException;
+use Psr\Container\ContainerInterface;
+
+/**
+ * Factory for PathResolver.
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class PathResolverFactory implements FactoryInterface
+{
+    /**
+     * Default config file directory under a local override directory
+     *
+     * @var string
+     */
+    protected $defaultConfigDirectory = PathResolver::DEFAULT_CONFIG_SUBDIR;
+
+    /**
+     * Create an object
+     *
+     * @param ContainerInterface $container     Service manager
+     * @param string             $requestedName Service being created
+     * @param null|array         $options       Extra options (optional)
+     *
+     * @return object
+     *
+     * @throws ServiceNotFoundException if unable to resolve the service.
+     * @throws ServiceNotCreatedException if an exception is raised when
+     * creating a service.
+     * @throws ContainerException&\Throwable if any other error occurs
+     */
+    public function __invoke(
+        ContainerInterface $container,
+        $requestedName,
+        array $options = null
+    ) {
+        if (!empty($options)) {
+            throw new \Exception('Unexpected options sent to factory.');
+        }
+        $localDirs = defined('LOCAL_OVERRIDE_DIR')
+            && strlen(trim(LOCAL_OVERRIDE_DIR)) > 0
+                ? [
+                    [
+                        'directory' => LOCAL_OVERRIDE_DIR,
+                        'defaultConfigSubdir' => $this->defaultConfigDirectory
+                    ]
+                ] : [];
+        return new $requestedName(
+            [
+                'directory' => APPLICATION_PATH,
+                'defaultConfigSubdir' => 'config/vufind'
```

Comment:
Another hard-coded 'config/vufind'. I can see arguments for keeping this distinct instead of using $this->defaultConfigDirectory... but at the same time, I can't really think of a scenario where you would want to change how the local directory works independently of the core directory...